### ISSUE TITLE
amp-lightbox-gallery: lightbox cannot be triggered or operated using the keyboard

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31161]
## WCAG Level
Level AA

## Priority
Critical

## Type
Interaction

## Pages/screens/components affected
* [amp-lightbox-gallery](https://amp.dev/documentation/components/amp-lightbox-gallery/)

## Description
Good design means supporting different modes of interaction, because there are many reasons why someone might use a keyboard instead of a mouse, and if keyboard interaction is not fully supported the website is effectively unusable.

Currently, the lightbox cannot be triggered using the keyboard. Even if a keyboard user managed to open the lightbox, none of the controls for the lightbox (the previous/next arrows, gallery, content, close control, and the thumbnails) are keyboard accessible.

![The main page, with images that trigger a lightbox](https://user-images.githubusercontent.com/54933393/99181346-c2c69700-2725-11eb-9062-1168950e05b3.png)

![The lightbox's content view](https://user-images.githubusercontent.com/54933393/99181349-ca863b80-2725-11eb-8188-5fefc08958c0.png)

![The lightbox's gallery view](https://user-images.githubusercontent.com/54933393/99181395-2355d400-2726-11eb-8164-8573f63c61c7.png)

## User impact
Someone with mobility problems will often use a keyboard instead of a mouse to navigate a web page. A screen reader user will also tend to use a keyboard if they cannot see the screen to use the mouse. Therefore, if the website does not properly support keyboard interaction a large number of people are effectively prevented from using it.

In this instance, keyboard users will not be able to use the component at all - they won't be able to get a larger/lightbox view of an image, nor will they be able to navigate through the lightbox itself.

## Required solution
Make sure that all interactive controls can be used with a keyboard.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance
Currently all lightbox controls are implemented as `<div>` elements. While they have an appropriate `role="button"`, and even an appropriate accessible name provided with `aria-label`, they are not keyboard-focusable.

```
<div role="button" class="..." data-action="close" aria-label="Close"></div>
```

We recommend using actual `<button>` elements for these controls.

```
<button ... aria-label="Close"></button>
```

Failing that, you can make the control keyboard-focusable by adding `tabindex="0"` - though you will then also need to provide explicit keyboard handling so that `Enter`/`Space` trigger the `click()` event.

The thumbnails in the gallery view, beyond not being keyboard-focusable, lack a `role` and text alternative/accessible name as well.

```
<div class="i-amphtml-lbg-gallery-thumbnail">
  <img class="..." srcset="...">
</div>
```

Once again, we recommend changing these to `<button>` elements (or adding the relevant `role`, `tabindex`, and keyboard handling script).

```
<button ... aria-label="View image 5">
  <img class="..." srcset="..." alt="Image 5">
</button>
```

Lastly, images in the main page that actually trigger the overall lightbox will also need to be keyboard focusable. Currently, they are simply `<amp-img>` elements with a `lightbox` attribute. Instead, these should also be actual `<button>` elements, similar to the thumbnails.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check that all interactive elements can be navigated to using the `Tab` key.
2. Check that all interactive elements can be activated using the `Enter` or `Space` keys.
3. Using a screen reader, also verify that the thumbnails/images that trigger the lightbox have an appropriate role and accessible name.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website  has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 2.1.1 Keyboard (Level A)](https://www.w3.org/TR/WCAG21/#keyboard)
* [WCAG 2.1 Success Criterion 4.1.2 Name, Role, Value (Level A)](https://www.w3.org/TR/WCAG21/#name-role-value)

## More information
* [WebAIM: Keyboard](https://webaim.org/techniques/keyboard/)
* [W3C - Understanding Success Criterion 4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
* [W3C - ARIA4: Using a WAI-ARIA role to expose the role of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4.html)

## Test data
Test date: November 2020
